File: permission/src/main/java/com/iwdael/onpermission/Permission.java
package com.iwdael.onpermission;

/**
 * author  : Iwdael
 * e-mail  : iwdael@outlook.com
 * github  : http://github.com/iwdael
 * project : OnPermission
 */
public interface Permission {
    String[] permissions();

    void onGranted(String[] permission);

    void onDenied(String[] permission);
}

File: annotation/src/main/java/com/iwdael/permissionsdispatcher/annotation/Permission.kt
package com.iwdael.permissionsdispatcher.annotation

/**
 * author : iwdael
 * e-mail : iwdael@outlook.com
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.BINARY)
annotation class Permission(val value: String, val min: Int = 23, val max: Int = Int.MAX_VALUE)
